File: LPS.py
string = input()
n = len(string)
dp =[]
for i in range(n):
    dp.append([0]*n)
    dp[i][i] = 1
i,j,k=0,1,1
while(k<n):
    if(string[i]==string[j]):
        dp[i][j] = dp[i+1][j-1]+2
    else:
        dp[i][j] = max(dp[i][j-1],dp[i+1][j])
    i+=1
    j+=1
    if(j==n):
        i=0
        k+=1
        j=k
print(dp[0][-1])

File: LPS.java
import java.util.*;
public class LPS
{
  public static void main(String[] args) {
    String str;
    Scanner sc = new Scanner(System.in);
    str = sc.nextLine();
    int n = str.length();
    int dp[][] = new int[n][n];
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n;j++)
        {
            if(i==j) dp[i][j]=1;
            else dp[i][j] = 0;
        }
    }
    int i=0,j=1,k=1;
    while(k<n)
    {
        if(str.charAt(i)==str.charAt(j))
        {
            dp[i][j] = dp[i+1][j-1]+2;
        }
        else
        {
            dp[i][j] = Math.max(dp[i][j-1],dp[i+1][j]);
        }
        i++;
        j++;
        if(j==n)
        {
            k++;
            i=0;
            j = k;
        }
    }
    System.out.print(dp[0][n-1]);
  }
}